### PR TITLE
Completed assignment week2

### PR DESCRIPTION
**Pull request by nos111**
Hi guys, Looking forward to your comments :)

### CODE REVIEW

File: week2/index.js [hunk 1]
```diff
@@ -1,65 +1,150 @@
+// require filesystem library because we are going to write and read files
 var fs = require('fs');
 
-var options = process.argv.slice(2);
+//this object will contain the todo's info and will be written to the todosFile 
+// I have chosen this to be a JSON because we can use the JSON proberities on it
+let todosFile = JSON.parse(fs.readFileSync('todo.json','UTF-8'));
+console.log(todosFile);
```
Comment (rts-rob) [inline week2/index.js]: Good for debugging, but did you mean to leave this in here for your submission?

This occurs no matter what the user enters, which is not in the spec.

File: week2/index.js [hunk 2]
```diff
@@ -1,65 +1,150 @@
...
+    //check if the user has entered text or not
+    if(itemToAdd === undefined || itemToAdd.length < 1){
+        console.log('Please enter a name for your task');
+    }else{
+        //push the text the user entered to our data file
+        jsonObject.tasks.push(itemToAdd);
+        //write the modified object to the todo File
+        console.log('Added new item');
+        jsonObject = JSON.stringify(jsonObject);
+        return fs.writeFile('todo.json',jsonObject);
```
Comment (rts-rob) [inline week2/index.js]: This results in:
`(node:5301) [DEP0013] DeprecationWarning: Calling an asynchronous function without callback is deprecated.`

Why does this happen and how can we fix it?

File: week2/index.js [hunk 3]
```diff
@@ -1,65 +1,150 @@
...
-      return console.log('Error: the help file could not be displayed', error);
+//add an item to the todo list
+//this functions takes the todoFile json and the commands array as parameters. 
+//It adds the new items to it and at the end it will write it the new object to the Todo file
+//this way we can save our new todos and keep our old ones
+function addTodoItem(jsonObject,commands) {
+    //assign the task text to a variable.
+    let itemToAdd = commands[1];
+    //check if the user has entered text or not
+    if(itemToAdd === undefined || itemToAdd.length < 1){
```
Comment (rts-rob) [inline week2/index.js]: Standard spacing (for greater clarity):
`if (itemToAdd === undefined || itemToAdd.length < 1) {`

File: week2/index.js [hunk 4]
```diff
@@ -1,65 +1,150 @@
...
+        console.log('Please enter a name for your task');
+    }else{
```
Comment (rts-rob) [inline week2/index.js]: Similarly: `} else {`

File: week2/index.js [hunk 5]
```diff
@@ -1,65 +1,150 @@
...
+    //checkc if the task number exists and makes sense
+    if(itemToRemove > jsonObject.tasks.length + 1) {
+        console.log('This item does not exist');
+    }else{
+        //remove the task from the jsonObject
+        jsonObject.tasks.splice(itemToRemove - 1,1)
+        console.log('Removed item number ' + itemToRemove)
+        //write the changed object to our Todofile to save the changes
+        jsonObject = JSON.stringify(jsonObject);
+        return fs.writeFile('todo.json',jsonObject);
```
Comment (rts-rob) [inline week2/index.js]: Same warning here as line 80

File: week2/index.js [hunk 6]
```diff
@@ -1,65 +1,150 @@
...
+//resets the amount of tasks to 0
+//takes the todosFile json and modifies the tasks in it
+//this function is invoked when the user enter the command 'reset'
+function resetTodosList (jsonObject) {
+    //empty tasks
+    jsonObject.tasks = [];
+    console.log('tasks list emptied successfully');
+    //turn into string and save it to the file
+    jsonObject = JSON.stringify(jsonObject);
+    return fs.writeFile('todo.json',jsonObject);
```
Comment (rts-rob) [inline week2/index.js]: Same warning here as line 80

File: week2/index.js [hunk 7]
```diff
@@ -1,65 +1,150 @@
...
+function updateTodosList(jsonObject,commands) {
+    let itemToUpdate = commands[1];
+    let newText = commands[2];
+    if(itemToUpdate > jsonObject.tasks.length) {
+        console.log('This item does not exist');
+    }else {
+        jsonObject.tasks[itemToUpdate - 1] = newText;
+        console.log('Updated item number '+itemToUpdate);
+        jsonObject = JSON.stringify(jsonObject);
+        return fs.writeFile('todo.json',jsonObject);
```
Comment (rts-rob) [inline week2/index.js]: Same warning here as line 80

File: week2/index.js [hunk 8]
```diff
@@ -1,65 +1,150 @@
...
-  default:
-    showHelp();
-    break;
-  case 'list':
-    listTodos();
-    break;
+    //if the user enter the word help as command
+    //we will display an introduction to all the commands available
+    //We will do it,through calling the functions showHelp
+    case 'help':
```
Comment (rts-rob) [inline week2/index.js]: What would happen if you moved this down to just above the `default` case and removed the `break`?

**Comment by rts-rob** [comment]
Overall looks good, and good job getting it turned in so quickly!

Have a look at my comments, especially line 80 and related. The warning gives you a good hint on how to resolve it.

Finally, don't be afraid of whitespace! Whether between braces `{ }` or parentheses `( )` or even between lines, it makes it easier to review and gain an understanding of your code more quickly.

File: exercies/content.md [hunk 9]
```diff
@@ -0,0 +1 @@
+'this file was created by nour in a train station where the fucking train didn't come'
```
Comment (jasongwartz) [inline exercies/content.md]: I'm going to leave a funny story about this in Slack.

File: week2/index.js [hunk 10]
```diff
@@ -1,65 +1,150 @@
...
 
-var options = process.argv.slice(2);
+//this object will contain the todo's info and will be written to the todosFile 
+// I have chosen this to be a JSON because we can use the JSON proberities on it
+let todosFile = JSON.parse(fs.readFileSync('todo.json','UTF-8'));
+console.log(todosFile);
 
-var command = options[0];
+//shorten the arguments array to the arguments we need
+//take out the first two arguments which are the node path and the file path
```
Comment (jasongwartz) [inline week2/index.js]: This is a great example of a "WHY" comment.

"Take out the first two arguments" is a WHAT.
"We don't care about the node path and the file path" is a WHY.

Keep writing comments like these 👍

File: week2/index.js (hunk 3 above)
Comment (jasongwartz) [inline week2/index.js]: +1 on this comment. Whitespace is cheap.

**Pull request closed without merging**